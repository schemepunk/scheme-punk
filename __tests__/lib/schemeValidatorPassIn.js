'use strict';

const Ajv = require('ajv');

const {
  schemeValidator,
  schemePunkSchema,
} = require('../../lib/schemeValidator');

describe('Scheme Runner Validation', async () => {
  test('Can pass in AJV Validator', async () => {
    expect.assertions(1);
    const testScheme = {
      source: {
        target: 'test',
        plugin: 'originalSchemeSource',
      },
      transform: {
        plugin: 'appendValuesAdapter',
        sourceAppend: '-',
      },
      destination: {
        target: 'test',
        plugin: 'concatIntoDestination',
      },
    };
    const ajv = new Ajv({allErrors: true});
    const schemePunkValidator = ajv.compile(schemePunkSchema);
    const validators = {schemePunkValidator};
    return expect(await
    schemeValidator({
      scheme: testScheme,
      useValidator: 'schemePunkValidator',
      validators,
    })).toMatchSnapshot();
  });
});
