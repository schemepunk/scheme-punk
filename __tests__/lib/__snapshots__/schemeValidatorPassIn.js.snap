// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scheme Runner Validation Can pass in AJV Validator 1`] = `
Object {
  "destination": Object {
    "plugin": "concatIntoDestination",
    "target": "test",
  },
  "source": Object {
    "plugin": "originalSchemeSource",
    "target": "test",
  },
  "transform": Object {
    "plugin": "appendValuesAdapter",
    "sourceAppend": "-",
  },
}
`;
